zokrates_core/src/ir/expression.rs

```diff
@@ -105,14 +105,27 @@ impl<T: Field> Add<LinComb<T>> for LinComb<T> {
 
     fn add(self, other: LinComb<T>) -> LinComb<T> {
```

Comment:
Any hints why just concatenating `self.0` and `other.0` would not work?

Reply:
Can you please give me an intuition on how to perform `add` and `sub` on a linear combinations that are represented internally as a `Vec` of pairs `(var, value)`? It seems that I missing something whilst trying to mimic the `BTreeMap`-based code.

Reply:
It seems that concatenation is working for addition, but I not sure that it would for a substraction.

Reply:
I believe for subtraction you could negate the coefficient of the second lincomb
`self.0.into_iter().chain(other.0.into_iter().map(|(var, coeff) (-var, coeff))).collect()` or something